binder/Passes/InterfacesPass.cs

```diff
@@ -56,9 +56,17 @@ void CreateInterfaceImplementation(Class @interface)
             var @base = new BaseClassSpecifier { Type = new TagType(@interface) };
             impl.Bases.Add(@base);
 
-            var methods = @interface.Declarations.Where(d => d is Method).Cast<Method>();
+            var methods = new List<Method>(@interface.Declarations.OfType<Method>());
+            foreach (var baseInterface in @interface.Bases)
+            {
+                methods.AddRange(baseInterface.Class.Declarations.OfType<Method>());
+            }
+
             foreach (var method in methods)
             {
+                if (method.Name == "__getObject")
```

Comment:
Think it might be better to check for `IsImplicit` here, which is only true in the case of synthesised/artificial methods, in case we have a need to add more such methods later to the interface.